cli/azd/internal/appdetect/appdetect.go

```diff
@@ -154,10 +154,22 @@ func (a AzureDepServiceBus) ResourceDisplay() string {
 
 type AzureDepEventHubs struct {
 	EventHubsNamePropertyMap map[string]string
+	FromDependency           FromDependency
 	UseKafka                 bool
```

Comment:
Maybe `UseKafka` can be deleted, since we added dependency type?

Reply:
Yes....it can be determined by specific dependency type later. Let's finalize this PR first, then I will deprecate this property.

Reply:
Thank you.

Reply:
After adding dependency type, we can check if the Azure Event Hubs uses kafka by checking if its `FromDependency` contains kafka related dependency, so for each time, we have to iterate over all `DependencyType` to determine it. I still prefer reserving this property `UseKafka` to indicate, what do you think about it?

Reply:
How about creating a func named `useKafka`, it won't cost too much time because it's just calculation with CPU and Memory, no much IO operation.

Reply:
Agree.